website/addons/evernote/views.py

```diff
@@ -0,0 +1,215 @@
+from flask import request
+import httplib as http
+
+from framework.auth.decorators import must_be_logged_in
+from framework.exceptions import HTTPError, PermissionsError
+
+from website.addons.base import generic_views
+from website.addons.evernote import utils
+from website.addons.evernote.serializer import EvernoteSerializer
+from website.oauth.models import ExternalAccount
+from website.util import permissions
+from website.project.decorators import (
+    must_be_contributor_or_public,
+    must_have_addon,
+    must_be_addon_authorizer,
+    must_not_be_registration,
+    must_have_permission
+)
+
+import ENML2HTML
+
+import logging
+logger = logging.getLogger(__name__)
+
+SHORT_NAME = 'evernote'
+FULL_NAME = 'Evernote'
+
+@must_be_logged_in
+def evernote_get_user_settings(auth):
+    """ Returns the list of all of the current user's authorized Evernote accounts """
+    serializer = EvernoteSerializer(user_settings=auth.user.get_addon('evernote'))
+    return serializer.serialized_user_settings
+
+
+@must_have_addon('evernote', 'node')
+@must_have_permission(permissions.WRITE)
+def evernote_get_config(node_addon, auth, **kwargs):
+    """API that returns the serialized node settings."""
+    # following from box addon:
+    # if node_addon.external_account:
+    #     refresh_oauth_key(node_addon.external_account)
+
+    #logger.debug(node_addon)
+    return {
+        'result': EvernoteSerializer().serialize_settings(node_addon, auth.user),
+    }
+
+@must_not_be_registration
+@must_have_addon('evernote', 'user')
+@must_have_addon('evernote', 'node')
+@must_be_addon_authorizer('evernote')
+@must_have_permission(permissions.WRITE)
+def evernote_set_config(node_addon, user_addon, auth, **kwargs):
+    """View for changing a node's linked Evernote folder."""
+    folder = request.json.get('selected')
+    serializer = EvernoteSerializer(node_settings=node_addon)
+
+    uid = folder['id']
+    path = folder['path']
+
+    node_addon.set_folder(uid, auth=auth)
```

Comment:
`node_addon.set_folder` should be modified to accept the JSON dict containing the `id` and `path`, which should allow you to remove the extra API calls I referenced in [another comment](https://github.com/CenterForOpenScience/osf.io/pull/4670/files#r67699265)